bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -72,9 +103,53 @@
     private static final String CONFIG_ALLOW_BASIC_AUTH = "allowBasicAuth";
     private static final String CONFIG_IMPLICIT_USER_ROLE = "implicitUserRole";
 
+    private static final int authCacheSize = 10;
+
     private boolean allowBasicAuth = false;
     private boolean implicitUserRole = true;
 
+    private HashMap<ByteBuffer, UserSecurityContext> authCache = new HashMap<ByteBuffer, UserSecurityContext>();
+    private byte[] rndBytes = getRndBytes();
+
+    private static byte[] getRndBytes() {
+        final Random rd = new Random();
+        byte[] rndBytes = new byte[32];
+        rd.nextBytes(rndBytes);
+        return rndBytes;
+    }
+
+    @Nullable
+    private ByteBuffer getCacheKey(String creds) {
+        try {
+            final MessageDigest md = MessageDigest.getInstance("SHA-256");
+            md.update(rndBytes);
+            return ByteBuffer.wrap(md.digest(creds.getBytes()));
```

Comment:
If I want to cast to String I'll have do encode the returned ``byte[]`` into an hex string.
I'd have to use/create a byte -> hexstr function which only adds overhead so if it's possible I'd like to keep the byte structure